src/UiPath.Workflow/Validation/CSharpExpressionValidator.cs

```diff
@@ -75,4 +85,60 @@ protected override string CreateReferenceCode(string types, string names, string
         GetSyntaxTreeForExpression(expressionText);
 
     protected override string GetTypeName(Type type) => CompilerHelper.GetTypeName(type);
+
+    /// <summary>
+    /// Checks if there are multiline comments, and if they are properly closed
+    /// Checks if there are any multiline text statements, and if they are properly closed
+    /// </summary>
+    /// <param name="code"></param>
+    /// <returns></returns>
+    private static bool AreCommentsBalanced(string code)
+    {
+        bool multilineCommentFaulted = false;
+        bool multilineTextFaulted = false;
+        bool isInsideQuotes = false;
+
+        for (int i = 0; i < code.Length - 1; i++)
+        {
+            if (code[i] == '@' && !isInsideQuotes)
+            {
+                multilineTextFaulted = true;
+            }
+            else if (code[i] == '"')
+            {
```

Comment:
Besides the escape issue above, there are some more advanced scenarios which are not really handled. Two that come to mind are:

- raw string literals
```
"""
    Lorem ipsum "dolor" sit amet,
        consectetur adipiscing elit.
"""
```

- string interpolation, like `$"Hi, { "foo" + "bar" }"` - where who knows what the users will do...

Reply:
The prevalidate does not make sure this is a true valid expression, but rather that it will not break my compilation of all expressions, resulting in the bug described above. Both the scenarios provided here will cause an isolated error, which is enough to pinpoint the actual error. The problem is when your expression actually breaks the compilation and it cannot determine properly where the issue is.